contracts/ReputationMiningCycle.sol

```diff
@@ -399,9 +409,15 @@ contract ReputationMiningCycle is ReputationMiningCycleStorage, PatriciaTreeProo
     bytes32[] memory siblings2
   ) public
   {
+    require(submissionWindowClosed(), "colony-reputation-mining-cycle-submissions-not-closed");
     require(index < disputeRounds[round].length, "colony-reputation-mining-index-beyond-round-length");
+
+    // We reward the submitter for this, so we better have an opponent so that someone gets slashed and can pay for it
+    uint256 opponentIdx = (index % 2 == 1 ? index-1 : index + 1);
+    require(opponentIdx < disputeRounds[round].length, "colony-reputation-mining-no-opponent-yet");
```

Comment:
I thought the comment would help explain this, but I guess not. This is checking that we have an opponent that we can punish later . `disputeRounds[round]` is zero indexed, so if the `opponentIdx` calculated is not less than the `length`, it's out-of-range and so doesn't exist (yet). This will occur if this function is called for the last submission currently in `disputeRounds` if the index of that submission is even. Or another way of putting that, the last submission if the total number of submissions in that round is odd.

Now I've typed this out, I suspect this is a holdover from when I thought I could do each dispute separately, pairwise. This is probably not necessary now, because no-one is actually slashed in this call.